source/lib/course-search/load-cached-courses.js

```diff
@@ -3,24 +3,20 @@
 import type {CourseType, TermType} from './types'
 import {loadTermsFromStorage} from './update-course-storage'
 import {COURSE_STORAGE_DIR} from './urls'
+import flatten from 'lodash/flatten'
 import RNFS from 'react-native-fs'
 
-export async function loadCachedCourses(): Array<CourseType> {
+export async function loadCachedCourses(): Promise<Array<CourseType>> {
 	const terms: Array<TermType> = await loadTermsFromStorage()
-	let courses: Array<CourseType> = []
-
-	for (let i = 0; i < terms.length; i++) {
-		let newCourses: Array<CourseType> = await loadTermCoursesFromStorage(
-			terms[i],
-		)
-		courses.push(...newCourses)
-	}
-	// console.log(courses)
-	return courses
+	const promises = terms.map(term => loadTermCoursesFromStorage(term))
+	const coursesByTerm = await Promise.all(promises)
+	return flatten(coursesByTerm)
 }
 
-async function loadTermCoursesFromStorage(term: TermType): Array<CourseType> {
-	return RNFS.readFile(COURSE_STORAGE_DIR + term.path).then(async (contents) => {
+function loadTermCoursesFromStorage(
+	term: TermType,
+): Promise<Array<CourseType>> {
+	return RNFS.readFile(COURSE_STORAGE_DIR + term.path).then(contents => {
```

Comment:
I almost feel like we should make a helper `readJsonFile` function that does this for us, similarly to how we have `fetchJson`. Up to you.